sky/clouds/coreweave.py

```diff
@@ -0,0 +1,199 @@
+"""CoreWeave Cloud."""
+import os
+import typing
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Union
+
+from sky import clouds
+from sky.adaptors import coreweave
+from sky.utils import annotations
+from sky.utils import registry
+from sky.utils import resources_utils
+
+if typing.TYPE_CHECKING:
+    from sky import resources as resources_lib
+    from sky.utils import volume as volume_lib
+
+
+def coreweave_profile_in_aws_cred_and_config() -> bool:
+    """Checks if CoreWeave Object Storage profile is set in aws credentials
+    and profile."""
+
+    credentials_path = os.path.expanduser('~/.aws/credentials')
+    coreweave_profile_exists_in_credentials = False
+    if os.path.isfile(credentials_path):
+        with open(credentials_path, 'r', encoding='utf-8') as file:
+            for line in file:
+                if f'[{coreweave.COREWEAVE_PROFILE_NAME}]' in line:
+                    coreweave_profile_exists_in_credentials = True
+
+    config_path = os.path.expanduser('~/.aws/config')
+    coreweave_profile_exists_in_config = False
+    if os.path.isfile(config_path):
+        with open(config_path, 'r', encoding='utf-8') as file:
+            for line in file:
+                if f'[profile {coreweave.COREWEAVE_PROFILE_NAME}]' in line:
+                    coreweave_profile_exists_in_config = True
+
+    return (coreweave_profile_exists_in_credentials and
+            coreweave_profile_exists_in_config)
+
+
+@registry.CLOUD_REGISTRY.register
+class CoreWeave(clouds.Cloud):
+    """CoreWeave Cloud"""
+    _REPR = 'CoreWeave'
+    _CLOUD_UNSUPPORTED_FEATURES = {
+        clouds.CloudImplementationFeatures.AUTODOWN:
+            ('Autodown not supported on CoreWeave.'),
+        clouds.CloudImplementationFeatures.SPOT_INSTANCE:
+            ('Spot instances not supported on CoreWeave.'),
+        clouds.CloudImplementationFeatures.CLONE_DISK_FROM_CLUSTER:
+            (f'Migrating disk is currently not supported on {_REPR}.'),
+        clouds.CloudImplementationFeatures.CUSTOM_DISK_TIER:
+            (f'Custom disk tier is currently not supported on {_REPR}.'),
+        clouds.CloudImplementationFeatures.CUSTOM_NETWORK_TIER:
+            ('Custom network tier is currently not supported on CoreWeave.'),
+        clouds.CloudImplementationFeatures.HIGH_AVAILABILITY_CONTROLLERS:
+            ('High availability controllers are not supported on CoreWeave.'),
+        clouds.CloudImplementationFeatures.CUSTOM_MULTI_NETWORK:
+            ('Customized multiple network interfaces are not supported on '
+             f'{_REPR}.'),
+    }
```

Comment:
Do we have to add clouds here? I thought we don't add it for R2 or Tigres?

Reply:
I thought it a bit more easier to add it as a cloud as its detected (by registry). There's no compute, so compute won't show up. 

I think since cloudflare already exists, I could switch it to be like how cloudflare was added

Reply:
Removed coreweave from `clouds`